### PR TITLE
Include the key in the "duplicate key" exception message

### PR DESCRIPTION
**Pull request by jkotas**
Port https://github.com/dotnet/coreclr/pull/1452 to CoreRT/.NET Native clone of Dictionary, as well as other collections implemented in System.Collections.dll.

### CODE REVIEW

**Comment by jkotas** [comment]
cc @weshaggard @akoeplinger

**Comment by weshaggard** [comment]
Thanks for porting this @jkotas. Doing a quick search for Argument_AddingDuplicate in corefx it looks like there might be a couple other places we have this. Would you mind grabbing those as well?

**Comment by jkotas** [comment]
@dotnet-bot test this please

**Comment by jkotas** [comment]
Fixed all instances of `Argument_AddingDuplicate`.

`System.Collections.NonGeneric` is using `Argument_AddingDuplicate__` that includes both key in dictionary and key being added. I left this one alone.

File: src/Common/src/System/Collections/Generic/LowLevelDictionary.cs [hunk 1]
```diff
@@ -20,9 +20,6 @@ namespace System.Collections.Generic
     ** behavior.)
     ** 
     ===========================================================*/
-#if TYPE_LOADER_IMPLEMENTATION
-    [System.Runtime.CompilerServices.ForceDictionaryLookups]
```
Comment (weshaggard) [inline src/Common/src/System/Collections/Generic/LowLevelDictionary.cs]: I assume you intentionally removed these and that they are dead code now?

**Comment by weshaggard** [comment]
LGTM

File: src/Common/src/System/Collections/Generic/LowLevelDictionary.cs (hunk 1 above)
Comment (jkotas) [inline src/Common/src/System/Collections/Generic/LowLevelDictionary.cs]: Yes, it is dead code for corefx.

**Pull request merged**